### ISSUE TITLE
Various RPC discrepancies compared to Lotus

### ISSUE THREAD

**Issue by alt-ivan** [ChainSafe/forest#4424]
### 1) `StateSectorPreCommitInfo`
When sector precommit info is not found/non-existent, the RPC returns an error instead of `null` (as Lotus does).
Expected (Lotus response):
```
{"jsonrpc":"2.0","result":null,"id":1}"
```

Actual (Forest):
```
{"jsonrpc":"2.0","error":{"code":-32603,"message":"SectorPreCommitOnChainInfo not found"},"id":1}
```

### 2) 'Message` deserialization has required fields that are optional in Lotus and default to 0 there

The fields
```
version
nonce
gas_limit
gas_fee_cap
gas_premium
```

are all optional when using Lotus (and will default to 0), whereas in Forest they are required and deserialization fails without them.

### 3) `StateWaitMsg` missing parameters

In Forest, this RPC only accepts two parameters and two are missing. 
Currently accepted parameters: 
```
message_cid
confidence
```

Missing parameters:
```
limit: ChainEpoch
allow_replaced: bool
```

To summarize, Lotus accepts all 4 in that exact order:
```
message_cid
confidence
limit
allow_replaced
```

## To reproduce

Call the related RPCs

## Log output

<details>
  <summary>Log Output</summary>
  
  ```Paste log output here
  paste log output...
  ```
</details>

## Environment (please complete the following information):

- OS:
- Branch/commit
- Hardware

**Comment by hanabi1224** [comment]
I will look into this

**Comment by hanabi1224** [comment]
Hey @alt-ivan 
Could you post the lotus version (`lotus --version`) you were testing against? And it will be great to have the payload you were using as well.

Regarding `Filecoin.StateSectorPreCommitInfo`, I get the below response when the sector number is invalid

```
# payload
{"jsonrpc":"2.0","id":0,"method":"Filecoin.StateSectorPreCommitInfo","params":
["f04040",18446744073709551615,[{"/":"bafy2bzacednxz2tzqrneijbboehcrcbdt2357musif4dyetzyyjlvu6q6xnpq"},{"/":"bafy2bzaceblbbv22npiwjhqta3dmxxpzyedjrfxugsell3ouprocd3x333bqo"}]]
}
# response
{"error":{"code":1,"message":"precommit info does not exist"},"id":0,"jsonrpc":"2.0"}
```

**Comment by alt-ivan** [comment]
Hi @hanabi1224 ,

I am using this lotus image via docker-compose (though I suppose it could be considered very old at this point): `ghcr.io/chainsafe/lotus-devnet:2024-04-07-d6d22ca`

If there is parity with the newer version, I will update our codebase accordingly, and please ignore the first item :)

**Comment by hanabi1224** [comment]
Update on `3) StateWaitMsg missing parameters`

@alt-ivan could you describe how you tested this RPC method with 3 or 4 parameter against lotus? I tried to supply more than 2 parameter and got below error

```
# payload
{"jsonrpc":"2.0","id":0,"method":"Filecoin.StateWaitMsg","params":
[{"/":"bafy2bzaceb46hxz3jzwkgrztleojhad2eruk5dhyrz3ejqbgiyw4hzcp2dr4i"},0, 100]
}

# response
{"error":{"code":-32602,"message":"wrong param count (method 'Filecoin.StateWaitMsg'): 3 != 2"},"id":0,"jsonrpc":"2.0"}
```

According to the [doc](https://github.com/filecoin-project/lotus/blob/master/documentation/en/api-v0-methods.md#statewaitmsg), it should accept exactly 2 parameter.
> [
  {
    "/": "bafy2bzacea3wsdh6y3a36tb3skempjoxqpuyompjbmfeyf34fi3uy6uue42v4"
  },
  42
]

For the `limit` parameter, there is another [`Filecoin.StateWaitMessageLimited`](https://github.com/filecoin-project/lotus/blob/master/documentation/en/api-v0-methods.md#statewaitmsglimited) yet to be implemented in forest.

**Comment by hanabi1224** [comment]
Update on `2) 'Message` deserialization has required fields that are optional in Lotus and default to 0 there`

Fixed in https://github.com/ChainSafe/forest/pull/4440

**Comment by hanabi1224** [comment]
@alt-ivan, thanks for clarifying! Yes, I was testing the V0 API. Let me see what I can do for the V1 PRC methods.

**Comment by hanabi1224** [comment]
@alt-ivan 1) and 3) have been fixed by #4443 and #4444 respectively.  Thanks for reporting!

**Comment by alt-ivan** [comment]
@hanabi1224 

I have tested and can confirm that the changes are working as expected! Thanks!

I will report tho other discrepancies I've found here. If you want me to open another issue, please let me know. Both pertain to V1 of the Lotus API.

1) `MpoolPending`
    * expected response: empty json array `result: []` if no messages found
    * actual response: `result: null`
2) 'StateSectorGetInfo`
    * expected response: `result: null` if sector commit info not found
    * actual response: 
        ```
        "error": {
        "code": -32603,
        "message": "Info for sector number 2 not found"
        },
        ```

**Comment by hanabi1224** [comment]
Hey @alt-ivan `MpoolPending` and `StateSectorGetInfo` should have been fixed by #4481 and #4482 respectively. Thanks for reporting, as always!

**Comment by alt-ivan** [comment]
@hanabi1224 
Thank you! I can confirm the changes are working as expected 🎉

**Comment by LesnyRumcajs** [comment]
@alt-ivan Great news. Feel free to open a new issue if anything else comes up.
